### PR TITLE
Enable pprof in proxy server

### PR DESCRIPTION
**Pull request by caesarxuchao**
Useful to debug memory leaks, etc.

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -519,6 +527,11 @@ func (p *Proxy) runAgentServer(o *ProxyRunOptions, server *server.ProxyServer) e
 func (p *Proxy) runAdminServer(o *ProxyRunOptions, server *server.ProxyServer) error {
 	muxHandler := http.NewServeMux()
 	muxHandler.Handle("/metrics", promhttp.Handler())
+	muxHandler.HandleFunc("/debug/pprof", pprof.Index)
```
Comment (cheftako) [inline cmd/server/main.go:530]: This is enabling a lot more than just contention profiling indicated by the flag.

```
Count | Profile
-- | --
0 | allocs
0 | block
0 | cmdline
12 | goroutine
0 | heap
0 | mutex
0 | profile
13 | threadcreate
0 | trace
```

File: cmd/server/main.go [hunk 2]
```diff
@@ -114,6 +119,7 @@ func (o *ProxyRunOptions) Flags() *pflag.FlagSet {
 	flags.UintVar(&o.agentPort, "agent-port", o.agentPort, "Port we listen for agent connections on.")
 	flags.UintVar(&o.adminPort, "admin-port", o.adminPort, "Port we listen for admin connections on.")
 	flags.UintVar(&o.healthPort, "health-port", o.healthPort, "Port we listen for health connections on.")
+	flags.BoolVar(&o.enableContentionProfiling, "enable-contention-profiling", o.enableContentionProfiling, "enable contention profiling at host:admin-port/debug/pprof/debug")
```
Comment (cheftako) [inline cmd/server/main.go:122]: For me the pprof is at http://host:admin-port/debug/pprof

Comment (caesarxuchao) [reply cmd/server/main.go:122]: Weird. For me it's debug/pprof/block, it's also the address in the [doc](https://golang.org/pkg/net/http/pprof/). debug/block returns 404 for me. Can you double check?

Comment (caesarxuchao) [reply cmd/server/main.go:122]: Oh I see /debug/pprof/debug is a typo. I fixed it.

File: cmd/server/main.go (hunk 1 above)
Comment (caesarxuchao) [reply cmd/server/main.go]: Good point. Added --enable-profiling.

**Comment by caesarxuchao** [comment]
@cheftako PTAL. Thanks.

File: cmd/server/main.go (hunk 2 above)
Comment (cheftako) [reply cmd/server/main.go]: Looks like we both typo'd that one :smile_cat:

File: cmd/server/main.go [hunk 3]
```diff
@@ -519,6 +535,13 @@ func (p *Proxy) runAgentServer(o *ProxyRunOptions, server *server.ProxyServer) e
 func (p *Proxy) runAdminServer(o *ProxyRunOptions, server *server.ProxyServer) error {
 	muxHandler := http.NewServeMux()
 	muxHandler.Handle("/metrics", promhttp.Handler())
+	if o.enableProfiling {
+		muxHandler.HandleFunc("/debug/pprof", pprof.Index)
```
Comment (cheftako) [inline cmd/server/main.go:539]: Not a blocker but I noticed that the links generated on the pprof index page when you hit /debug/pprof are for /debug/X not for /debug/pprof/X. So you get 404's if you click any of the links. The links are fine if you hit /debug/pprof/. You might consider using a redirect on "/debug/pprof" to send them to the good url.

func indexRedirect(w http.ResponseWriter, r *http.Request) {
	http.Redirect(w, r, "/debug/pprof/", http.StatusMovedPermanently)
}

Comment (caesarxuchao) [reply cmd/server/main.go]: Done.  PTAL. Thanks.

**Pull request merged**